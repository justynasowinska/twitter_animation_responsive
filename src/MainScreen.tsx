import React, { useState, useContext } from 'react';
import { StyleSheet, View, ScrollView, Animated, ViewStyle } from 'react-native';
import { Header } from './Header';
import { Avatar } from './Avatar';
import { images } from './utils/images';
import { getAnimatedValues } from './animations/Animations';
import { ScreenSizeContext } from './context/ScreenSizeContext';
import { getHeightForElements } from './animations/helpers';

export const MainScreen = () => {
  const [scrollY] = useState(new Animated.Value(0));
  const { height } = useContext(ScreenSizeContext);

  const mock_username = 'Justyna_S';

  const {
    headerHeight,
    imageSize,
    imageMarginTop,
    headerZIndex,
    headerTitleBottom,
  } = getAnimatedValues(scrollY, height);

  const { headerFontSize, avatarFontSize } = getHeightForElements(height);

  return (
    <>
      <Header
          userName={mock_username}
          animatedHeaderStyles={{ height: headerHeight, zIndex: headerZIndex }}
          animatedTextStyles={{ bottom: headerTitleBottom }}
          userNameFontSize={headerFontSize}
      />
      <ScrollView
          style={styles.scrollView}
          scrollEventThrottle={16}
          onScroll={
              Animated.event([
                  { nativeEvent: { contentOffset: { y: scrollY } } }
              ])
          }
      >
          <Avatar
              image={images.avatarImg}
              username={mock_username}
              containerStyles={styles.avatarContainer}
              userNameFontSize={avatarFontSize}
              animatedStyles={{ width: imageSize, height: imageSize, marginTop: imageMarginTop }}
          />
          <View style={{ height: height * 2 }} />
      </ScrollView>
    </>
  );
};

interface Style {
    scrollView: ViewStyle;
    avatarContainer: ViewStyle;
}

const styles = StyleSheet.create<Style>({
  scrollView: {
    flex: 1
  },
  avatarContainer: {
      marginLeft: '3%'
  }
});
