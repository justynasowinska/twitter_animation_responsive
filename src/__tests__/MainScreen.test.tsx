import React from 'react';
import { shallow } from 'enzyme';

import { MainScreen } from '../MainScreen';

describe('MainScreen component', () => {
    it('initially renders correctly', () => {
        const component = shallow(<MainScreen />);
        expect(component).toMatchSnapshot();
    });
});