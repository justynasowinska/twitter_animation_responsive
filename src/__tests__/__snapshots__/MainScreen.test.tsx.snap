// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen component initially renders correctly 1`] = `
<Fragment>
  <Header
    animatedHeaderStyles={
      Object {
        "height": 213.44,
        "zIndex": 0,
      }
    }
    animatedTextStyles={
      Object {
        "bottom": -54.3605,
      }
    }
    userName="Justyna_S"
    userNameFontSize={29.348}
  />
  <ScrollViewMock
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="scrollViewComponent"
  >
    <Avatar
      animatedStyles={
        Object {
          "height": 133.4,
          "marginTop": 146.74,
          "width": 133.4,
        }
      }
      containerStyles={
        Object {
          "marginLeft": "3%",
        }
      }
      image={
        Object {
          "testUri": "../../../assets/images/me.jpeg",
        }
      }
      userNameFontSize={36.018}
      username="Justyna_S"
    />
    <View
      style={
        Object {
          "height": 2668,
        }
      }
    />
  </ScrollViewMock>
</Fragment>
`;
